astroid/raw_building.py

```diff
@@ -116,20 +116,22 @@ def build_module(name: str, doc: Optional[str] = None) -> nodes.Module:
     return node
 
 
-def build_class(name, basenames=(), doc=None):
-    """create and initialize an astroid ClassDef node"""
+def build_class(
+    name: str, basenames: Sequence[str] = (), doc: Optional[str] = None
```

Comment:
```suggestion
    name: str, basenames: Iterable[str] = (), doc: Optional[str] = None
```
If you start that way, we should use the least limiting base class. 😉